pkg/smokescreen/smokescreen.go

```diff
@@ -687,13 +689,17 @@ func BuildProxy(config *Config) *goproxy.ProxyHttpServer {
 		// In the context of MITM request, reuse the existing SmokescreenContext (which has the role from CONNECT)
 		// but perform ACL check on the new destination
 		if pctx.ConnectAction == goproxy.ConnectMitm {
-			if existingSctx, ok := pctx.UserData.(*SmokescreenContext); ok {
-				config.Log.Info("MITM request, reusing role from CONNECT but checking new destination")
-				sctx = existingSctx
-			} else {
-				config.Log.Error("MITM request but UserData is not SmokescreenContext!")
-				sctx = newContext(config, httpProxy, req)
+			existingSctx, ok := pctx.UserData.(*SmokescreenContext)
+			if !ok || existingSctx.Decision == nil {
+				config.Log.Error("MITM request but UserData is not SmokescreenContext or missing Decision - rejecting request")
```

Comment:
Nit: there are 2 branches that will get us here. If we ever have to debug this, it will be very useful to understand in which branch we are 
```suggestion
				config.Log.WithFields(logrus.Fields{
					"context_valid": ok,
					"decision_present": existingSctx != nil && existingSctx.Decision != nil,
				}).Error("MITM request missing required context or decision from CONNECT phase - rejecting request")
```